### ISSUE TITLE
namespace config is not set in umbrella project

### ISSUE THREAD

**Issue by chvanikoff** [phoenixframework/phoenix#3019]
* Elixir version (elixir -v): 1.7.2
* Phoenix version (mix deps): 1.3.4

### Expected behavior

In project generated with `mix phx.new testapp --umbrella --module=Test`, following config should present under `apps/testapp/config/config.exs` and `apps/testapp_web/config/config.exs`:
```
config :testapp, namespace: Test
```

### Actual behavior

The config only presents under `apps/testapp_web/config/config.exs`, but absent at `apps/testapp/config/config.exs`

**Comment by chrismccord** [comment]
The config is used only for web-facing modules/values as far as I remember. Can you point to a place where you are seeing incorrectly named modules, and which generator commands you ran? If it's simply  that the config line isn't present in the other app, then that is not an issue. Thanks!

**Comment by chvanikoff** [comment]
Yes, there seems to be inconsistency in module naming when using generators: i.e. under `apps/testapp` app module is named `Test.Application` while generated with any of `mix phx.gen.{context,html,json,...} Accounts User users name` would be `Testapp.User`.
And even if it was intended, there's still inconsistency when generating tests: i.e., generated files from example above contain `use Testapp.DataCase` while the `DataCase` module is defined within `Test` namespace from `phx.new`

**Comment by mikeover** [comment]
@chvanikoff #3031 look good to you?
